airflow/models.py

```diff
@@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
             DR.dag_id == self.dag_id,
             DR.execution_date == self.execution_date,
             DR.run_id == self.run_id
-        ).one()
+        ).first()
```

Comment:
It does. When you self.dag_id, self.execution_date, and self.run_id are not present anymore in the db it should throw an exception. This is an integrity issue. I think it should be reverted to the old behavior. Airflow should throw an exception when the dag_run is not present anymore in the db while it still has it in memory.